actions/databases.go

```diff
@@ -0,0 +1,290 @@
+		}
+
+		clusterOutput, err := rdsClient.Service.CreateDBCluster(input.Cluster)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				log.Println(aerr.Error())
+				return c.Error(400, aerr)
+			}
+			log.Println(err.Error())
+			return err
+		}
+		log.Println("Created RDS cluster", clusterOutput)
+	}
+
+	// create rds instance
+	if input.Instance.DBSubnetGroupName == nil {
+		input.Instance.DBSubnetGroupName = aws.String(rdsClient.DefaultSubnet)
+	}
+
+	instanceOutput, err := rdsClient.Service.CreateDBInstance(input.Instance)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			log.Println(aerr.Error())
+			return c.Error(400, aerr)
+		}
+		log.Println(err.Error())
+		return err
+	}
+	log.Println("Created RDS instance", instanceOutput)
+
+	return c.Render(200, r.JSON(instanceOutput))
+}
+
+// DatabasesDelete deletes a database in a given account
+//   It will delete the database instance with the given {db} name and will also delete the associated cluster
+//   if the instance belongs to a cluster and is the last remaining member.
+//   If the snapshot=true parameter is given, it will create a final snapshot of the instance/cluster.
+func DatabasesDelete(c buffalo.Context) error {
+	// if snapshot param is given, a final snapshot will be created before deleting
+	snapshot := false
+	if b, err := strconv.ParseBool(c.Param("snapshot")); err == nil {
+		snapshot = b
+	}
+
+	rdsClient := RDS[c.Param("account")]
+	var instanceInput *rds.DeleteDBInstanceInput
+	var clusterInput *rds.DeleteDBClusterInput
+
+	// first, let's determine if the given database instance belongs to a cluster
+	var clusterName *string
+	i, err := rdsClient.Service.DescribeDBInstances(&rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: aws.String(c.Param("db")),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case rds.ErrCodeDBInstanceNotFoundFault:
+				log.Println(rds.ErrCodeDBInstanceNotFoundFault, aerr.Error())
+				return c.Error(404, aerr)
+			default:
+				log.Println(aerr.Error())
+				return c.Error(400, aerr)
+			}
+		} else {
+			log.Println(err.Error())
+			return err
+		}
+	}
+	if i.DBInstances[0].DBClusterIdentifier != nil {
+		clusterName = i.DBInstances[0].DBClusterIdentifier
+	}
+
+	if snapshot && clusterName == nil {
+		log.Printf("Deleting database %s and creating final snapshot", c.Param("db"))
+		instanceInput = &rds.DeleteDBInstanceInput{
+			DBInstanceIdentifier:      aws.String(c.Param("db")),
+			FinalDBSnapshotIdentifier: aws.String("final-" + c.Param("db")),
+			SkipFinalSnapshot:         aws.Bool(false),
+		}
+	} else {
+		log.Printf("Deleting database %s without creating final snapshot", c.Param("db"))
+		instanceInput = &rds.DeleteDBInstanceInput{
+			DBInstanceIdentifier: aws.String(c.Param("db")),
+			SkipFinalSnapshot:    aws.Bool(true),
+		}
+	}
+
+	instanceOutput, err := rdsClient.Service.DeleteDBInstance(instanceInput)
```

Comment:
how i learned to stop worrying and love the `context`

Reply:
i assume i should be using `WithContext` for all the calls to RDS